src/sentry/profiles/task.py

```diff
@@ -86,8 +86,8 @@ def process_profile(
         return
 
     _initialize_publisher()
-    _insert_eventstream_call_tree(profile=profile)
     _insert_eventstream_profile(profile=profile)
+    _insert_eventstream_call_tree(profile=profile)
```

Comment:
The max recusion depth was hit inside of `_insert_vroom_profile`, so this task exits early on line 806. I think what we have to do here is the swallow any error that happens after we get a 200 response from vroom (this should mean that the profile was written to gcs), and make sure that `_insert_eventstream_call_tree` can handle the call tree not existing.

Reply:
Ah yes, but that won't really work as we're `json.loads`ing the response and it's likely what gave the error. If we swallow this error, we won't be able to read the call trees. So in the end, the only solution would be to limit the stack or pass the call tree response as is.